src/app/util/attribute-storage.h

```diff
@@ -378,5 +378,15 @@ bool IsFlatCompositionForEndpoint(EndpointId endpoint);
  */
 bool IsTreeCompositionForEndpoint(EndpointId endpoint);
 
+/**
+ * @brief Returns whether the composition is full family pattern for a given endpoint index
```

Comment:
How about a `GetCompositionForIndex` that returns an enum instead, so that we know all possible values? 

With two boolean methods it seems hard to handle "what if both are true?" or "what if both are false?"

Reply:
Replaced with a `GetCompositionForEndpointIndex` function.